src/BaseControllerV2.test.ts

```diff
@@ -511,4 +571,149 @@ describe('getPersistentState', () => {
+          return { visitors: [...visitors, visitor] };
+        });
+      }
+
+      destroy() {
+        super.destroy();
+      }
+    }
+
+    type VisitorOverflowControllerState = {
+      maxVisitors: number;
+    };
+    type VisitorOverflowControllerAction = {
+      type: `VisitorOverflowController:updateMax`;
+      handler: (max: number) => void;
+    };
+    type VisitorOverflowControllerEvent = {
+      type: `VisitorOverflowController:stateChange`;
+      payload: [VisitorOverflowControllerState, Patch[]];
+    };
+
+    const visitorOverflowControllerMetadata = {
+      maxVisitors: {
+        persist: false,
+        anonymous: true,
+      },
+    };
+
+    class VisitorOverflowController extends BaseController<
+      'VisitorOverflowController',
+      VisitorOverflowControllerState
+    > {
+      constructor(
+        messagingSystem: RestrictedControllerMessenger<
+          'VisitorOverflowController',
+          VisitorControllerAction | VisitorOverflowControllerAction,
+          VisitorControllerEvent | VisitorOverflowControllerEvent,
+          'VisitorController:clear',
+          'VisitorController:stateChange'
+        >,
+      ) {
+        super({
+          messenger: messagingSystem,
+          metadata: visitorOverflowControllerMetadata,
+          name: 'VisitorOverflowController',
+          state: { maxVisitors: 5 },
+        });
+        messagingSystem.registerActionHandler('VisitorOverflowController:updateMax', this.updateMax);
+        messagingSystem.subscribe('VisitorController:stateChange', this.onVisit);
+      }
+
+      onVisit = ({ visitors }: VisitorControllerState) => {
+        if (visitors.length > this.state.maxVisitors) {
+          this.messagingSystem.call('VisitorController:clear');
+        }
+      };
+
+      updateMax = (max: number) => {
+        this.update(() => {
+          return { maxVisitors: max };
+        });
+      };
+
+      destroy() {
+        super.destroy();
+      }
+    }
+
+    it('should allow messaging between controllers', () => {
+      const controllerMessenger = new ControllerMessenger<
+        VisitorControllerAction | VisitorOverflowControllerAction,
+        VisitorControllerEvent | VisitorOverflowControllerEvent
+      >();
+      const visitorControllerMessenger = controllerMessenger.getRestricted({
+        name: 'VisitorController',
+        allowedActions: [],
+        allowedEvents: [],
+      });
+      const visitorController = new VisitorController(visitorControllerMessenger);
+      const visitorOverflowControllerMessenger = controllerMessenger.getRestricted({
+        name: 'VisitorOverflowController',
+        allowedActions: ['VisitorController:clear'],
+        allowedEvents: ['VisitorController:stateChange'],
+      });
+      const visitorOverflowController = new VisitorOverflowController(visitorOverflowControllerMessenger);
+
+      controllerMessenger.call('VisitorOverflowController:updateMax', 2);
```

Comment:
Not a strong opinion, but I slightly feel like the method params should be an array as a second argument. I don't feel this strongly enough to request you do it, let alone block, but feel it just enough to want to record that twinge here in case anyone else feels more strongly.

Reply:
This is interesting :thinking: It would have been slightly easier for me to get the types working if this accepted an array. I had put the extra effort into unwrapping them because I thought nesting arguments in an array would make this more difficult to use. This felt to me more like a "normal" method.

It also allows us to do this:

```suggestion
      const updateMax = controllerMessenger.bind(controllerMessenger, 'VisitorOverflowController:updateMax');
      updateMax(2);
```
So that we can create an instance of the action that is _exactly_ like the function at the other end, except that it's facilitated by the messenger system.